### PR TITLE
add initial docs

### PR DESCRIPTION
**Pull request by jrhizor**
Check it out using the Stoat docs preview!

There's definitely a ton more we can do for docs, especially around explaining the `debug` page and how the merging happens, but I'm hoping this is enough to get started.

### CODE REVIEW

**Review by tuliren — APPROVED** [review]
Looks great!

Two suggestions:
- Use the `/docs/intro` page as the root page, so that people don't need to click the "Get Started!" button. The current root page is pretty empty.
- Add a few "Quick Start for X" pages.
  - I think right now people need to read a lot of text to start using our product. Even after following the "Setting Up Stoat", no plugin is activated.
  - The "Quick Start for X" pages do not need to include any explanation, just very simple steps to let user start using Stoat for specific use cases.
  - Here is an example of "Quick Start for Storybook":
    - Step 1. Install the Stoat App.
    - Step 2. Add the Stoat Action at the end of the job that builds the storybook.
      - Show the action snippet.
    - Step 3. Add the Stoat config file.
      - Show the config file.
    - That's it.
  - Other pages include:
    - Quick start for junit test report.
    - Quick start for docusaurus.
    - Quick start for jest test report.
  - I can actually write them after this PR is merged.
  - What do you think?

**Comment by tuliren** [comment]
The doc preview is beautiful! Imagine if people can comment on them directly.

File: docs/docs/tutorials/templating.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+When rendering PR comments, Stoat uses a [Handlebars](https://handlebarsjs.com/) template.
+Which template to use is specified by the optional `comment_template_file` key/value in the `.stoat/config.yaml` file.
+If this file is absent, a default template is used. A description of the [current default remote v1 template](https://www.stoat.dev/api/templates?stoatConfigVersion=1) is available via our API.
+
+## Modifying the comment template
+
+For this tutorial, let's first add a line of text that says "Hello World" at the output of the Stoat PR comment.
+
+Create `.stoat/template.hbs` with the following contents:
+```
```
Comment (tuliren) [inline docs/docs/tutorials/templating.md:12]: Nitpick.

```suggestion
```hbs
```

**Comment by jrhizor** [comment]
Agreed on the Quick Start-style guides. Let's merge this now and add those separately.

I'll also add a homepage redirect separately.

**Pull request merged**